src/providers/ipa/ipa_subdomains_id.c

```diff
@@ -312,6 +366,30 @@ static void ipa_subdomain_account_done(struct tevent_req *subreq)
         return;
     }
 
+    if (state->mapped_attrs != NULL) {
+        object_dom = sss_get_domain_by_sid_ldap_fallback(state->domain,
+                                                         state->object_sid);
+        ret = sysdb_search_object_by_sid(state,
+                                         object_dom != NULL ? object_dom
+                                                            : state->domain,
+                                         state->object_sid, NULL, &res);
+        if (ret == EOK) {
+            ret = sysdb_set_entry_attr(state->domain->sysdb, res->msgs[0]->dn,
```

Comment:
@sumit-bose, a simple question here ... above you fire the sysdb_search_object_by_sid() on the object_dom, in case it exists. Although, here you set the entry directly in the state->domain ... not caring about the object_dom (even i the search done before was done using it). Is this expected or shall we also consider object_dom here? And in case it's expected ... why?

Reply:
@fidencio, thank you for the careful review. sysdb_set_entry_attr() only needs the sysdb_ctx because the object is given by the DN. The sysdb_ctx is the same for all domains know to the backend because it is basically the handle to the cache ldb file where all data is stored. So, the sysdb component of any domain could be used here.

The domain attribute in the sysdb search calls is typically used to restrict the search base but this is not needed if you directly access the object with its DN.

Reply:
Thanks for the explanation, it does make sense.